vcd/resource_vcd_vm_affinity_rule.go

```diff
@@ -0,0 +1,387 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+func resourceVcdVmAffinityRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdVmAffinityRuleCreate,
+		Read:   resourceVcdVmAffinityRuleRead,
+		Update: resourceVcdVmAffinityRuleUpdate,
+		Delete: resourceVcdVmAffinityRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdVmAffinityRuleImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "VM affinity rule name",
+			},
+			"rule_id": &schema.Schema{
```

Comment:
why we need this? as this store Id in `setId`

Reply:
It's computed. I make it explicit because it's visible in the data source, where it becomes the preferred way of retrieving the entity, although search by name is also supported.

Reply:
I leave this to @lvirbalas to decide :)

Reply:
@dataclouder Correct me if I'm wrong, but we don't have this approach anywhere else. I don't understand why we need explicit `rule_id` if it's already accessible via the `id` field. From the first glance, looks a duplicate.

Reply:
We use it in other resources and data sources (e.g. nsxv_firewall_rule, nsxv_dnat), and it helps conveying to the user the idea that the name of the entity could be duplicated and the entity ID is needed. By making it explicit, we reinforce the concept that the

Reply:
just for info: nsxv_dnat has `rule_id` in datasource, but not in resource.

Reply:
> We use it in other resources and data sources (e.g. nsxv_firewall_rule, nsxv_dnat)

@dataclouder Yours is the only resource that has it. Only data sources have it so far:

![Screenshot 2020-06-19 at 08 56 58](https://user-images.githubusercontent.com/11330689/85101327-d7504300-b20a-11ea-983d-fdfc4777d96e.png)

> and it helps conveying to the user the idea that the name of the entity could be duplicated and the entity ID is needed

Please paste an example use case.

> By making it explicit, we reinforce the concept that the

Sentence ending lost :)

Reply:
Removed `rule_id` from resource